src/Symfony/Component/Validator/Constraints/DateValidator.php

```diff
@@ -61,7 +61,9 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
-        if (!self::checkDate($matches[1], $matches[2], $matches[3])) {
+        if (!isset($matches['year'], $matches['month'], $matches['day'])
+            || !self::checkDate($matches['year'], $matches['month'], $matches['day'])
```

Comment:
this breaks classes that override the constant. This would fix it:
`self::checkDate($matches['year'] ?? $matches[1], etc`

Reply:
Yes that’s why I added a note to UPGRADE, didn’t think that through, fixed!